### PR TITLE
Added logsoftmax BF16/FP32 FWD oneDNN kernel

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
New features

### PR changes
OPs

### Describe
Added logsoftmax BF16/FP32 FWD oneDNN kernel with tests in both old and new Paddle's UT frameworks

### CODE REVIEW

**Comment by jakpiase** [comment]
@piotrekobiIntel please review this PR

File: paddle/fluid/operators/log_softmax_op.cc [hunk 1]
```diff
@@ -31,9 +31,17 @@ class LogSoftmaxOp : public framework::OperatorWithKernel {
...
+        framework::OperatorWithKernel::IndicateVarDataType(ctx, "X");
+
+#ifdef PADDLE_WITH_MKLDNN
+    if (this->CanMKLDNNBeUsed(ctx, input_data_type)) {
+      return framework::OpKernelType(input_data_type, ctx.GetPlace(),
+                                     framework::DataLayout::kMKLDNN,
+                                     framework::LibraryType::kMKLDNN);
+    }
+#endif
+    return framework::OpKernelType(input_data_type, ctx.GetPlace());
```
Comment (ghost) [inline paddle/fluid/operators/log_softmax_op.cc:44]: The method used to return framework::OpKernelType(input_data_type, ctx.device_context()) when mkldnn was not used, but now returns framework::OpKernelType(input_data_type, ctx.GetPlace()). Is there a difference between ctx.device_context() and ctx.GetPlace()?

Comment (piotrekobi) [reply paddle/fluid/operators/log_softmax_op.cc:44]: I'm the ghost :)

Comment (jakpiase) [reply paddle/fluid/operators/log_softmax_op.cc:44]: @piotrekobi yes, there is a difference between `ctx.device_context()` and `ctx.GetPlace()`, but in this particular case `OpKernelType`'s constructor is overloaded and we achieve identical result whether we pass `ctx.device_context()` or `ctx.GetPlace()`. Code that is responsible for that is linked below
https://github.com/PaddlePaddle/Paddle/blob/058e1d8592e44fc913365520455d124333224adf/paddle/fluid/framework/operator.h#L389

https://github.com/PaddlePaddle/Paddle/blob/058e1d8592e44fc913365520455d124333224adf/paddle/fluid/framework/op_kernel_type.h#L39-L58

**Review by piotrekobi — APPROVED** [review]
LGTM

**Review by jczaja — APPROVED** [review]
LGTM.

**Comment by jczaja** [comment]
@baoachun Please start your review

**Review by baoachun — APPROVED** [review]
LGTM

**Pull request merged**